sunpy/data/sample.py

```diff
@@ -1,35 +1,62 @@
 """
-This module provides the following sample data files.  These files are
-downloaded when this module is imported for the first time.
+This module provides the following sample data files.  When a sample shortname
+is accessed, the corresponding file is downloaded if needed.  All files can be
+downloaded by calling :func:`~sunpy.data.sample.download_all`.
 
+Summary variables
+-----------------
+.. list-table::
+   :widths: auto
+
+   * - ``file_dict``
+     - Dictionary of all sample shortnames and, if downloaded, file locations
+       on disk
+   * - ``file_list``
+     - List of disk locations for sample data files that have been downloaded
+
+Sample shortnames
+-----------------
 .. list-table::
    :widths: auto
    :header-rows: 1
 
-   * - Variable name
+   * - Sample shortname
      - Name of downloaded file
 """
-import sys
-from pathlib import Path
+from ._sample import _SAMPLE_DATA, _get_sample_files, download_sample_data
+
+# Add a table row to the module docstring for each sample file
+for _keyname, _filename in sorted(_SAMPLE_DATA.items()):
+    __doc__ += f'   * - ``{_keyname}``\n     - {_filename}\n'
+
+
+__all__ = list(sorted(_SAMPLE_DATA.keys())) + ['download_all', 'file_dict', 'file_list']
 
-from ._sample import _SAMPLE_FILES, download_sample_data
 
-files = download_sample_data()
+def __dir__():
+    return __all__
```

Comment:
Consider adding a comment that links to [PEP 562](https://peps.python.org/pep-0562/), the proposal for allowing module-level `__dict__` and `__getattr__`. This is esoteric enough that a clarifying comment and helpful link would be beneficial to future readers.

Reply:
Done